spec/controllers/projects/contributions_controller_spec.rb

```diff
@@ -110,6 +110,25 @@
     end
   end
 
+  describe 'unavailable contribution projects' do
+    let(:project) { create(:project, available_for_contribution: false) }
+    let(:reward)  { create(:reward, project: project) }
+
+    context 'new' do
```

Comment:
What I'm talking about is that looking the code, this case that you are testing here on the new context can happen on the post context. The inverse is true too, what is being tested inside the post context can happen here. For this reason, it appear to me that you're not testing the cases completely